src/main/java/org/opensearch/jobscheduler/rest/action/RestReleaseLockAction.java

```diff
@@ -0,0 +1,140 @@
+
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.jobscheduler.JobSchedulerPlugin;
+import org.opensearch.jobscheduler.spi.LockModel;
+import org.opensearch.jobscheduler.spi.utils.LockService;
+import org.opensearch.jobscheduler.utils.JobDetailsService;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestRequest;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+import org.opensearch.rest.RestStatus;
+
+public class RestReleaseLockAction extends BaseRestHandler {
+
+    public static final String RELEASE_LOCK_ACTION = "release_lock_action";
+    public static final String LOCK_ID = "lock_id";
+
+    private final Logger logger = LogManager.getLogger(RestReleaseLockAction.class);
+
+    private LockService lockService;
+
+    private LockModel releaseLock;
+
+    private boolean releaseResponse;
+
+    public RestReleaseLockAction(LockService lockService) {
+        this.lockService = lockService;
+    }
+
+    @Override
+    public String getName() {
+        return RELEASE_LOCK_ACTION;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ImmutableList.of(
+            new Route(PUT, String.format(Locale.ROOT, "%s/%s/{%s}", JobSchedulerPlugin.JS_BASE_URI, "_release_lock", LOCK_ID))
+        );
+    }
+
+    @Override
+    public RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient nodeClient) throws IOException {
+        String lockId = restRequest.param(LOCK_ID);
+        if (lockId == null || lockId.isEmpty()) {
+            throw new IOException("lockId cannot be null or empty");
+        }
+
+        CompletableFuture<LockModel> findInProgressFuture = new CompletableFuture<>();
+        lockService.findLock(lockId, ActionListener.wrap(lock -> {
+            releaseLock = lock;
+            findInProgressFuture.complete(lock);
+        }, exception -> {
+            logger.error("Could not find lock model with lockId " + lockId, exception);
+            findInProgressFuture.completeExceptionally(exception);
+        }));
+
+        try {
+            findInProgressFuture.orTimeout(JobDetailsService.TIME_OUT_FOR_REQUEST, TimeUnit.SECONDS).join();
+        } catch (CompletionException e) {
+            if (e.getCause() instanceof TimeoutException) {
+                logger.info(" Request timed out with an exception ", e);
+            } else {
+                throw e;
```

Comment:
Hmm.. do we need the else ? And throw the exception? 
The catch all at 87 will anyway take care of it.

Reply:
@dbwiddis suggested to catch time out exception seperately

Reply:
Ok if so we need need handle this cleanly, would it drop the rest request ? 
Is there a test which tests this case?

Reply:
No as of now there is no test for this

Reply:
Lets add one.

Reply:
OK, I think I may have been confused before and reviewing a PR on opensearch cleared it up.  We do need to clean up this exception handling.  Take a look at `ExternsionsManager` and try to follow this pattern:
```java
} catch (CompletionException e) {
    if (e.getCause() instanceof TimeoutException) {
        logger.info("No response from extension to request.");
    }
    if (e.getCause() instanceof RuntimeException) {
        throw (RuntimeException) e.getCause();
    } else if (e.getCause() instanceof Error) {
        throw (Error) e.getCause();
    } else {
        throw new RuntimeException(e.getCause());
    }
}
```